src/frontend/src/optimizer/plan_node/mod.rs

```diff
@@ -697,6 +703,55 @@ impl Explain for PlanRef {
         let explain_ir = plan.explain();
         serde_yaml::to_string(&PrettySerde(explain_ir)).expect("failed to serialize plan to yaml")
     }
+
+    /// Explain the plan node and return a dot format string.
+    fn explain_to_dot(&self) -> String {
+        let plan = reorganize_elements_id(self.clone());
+        let explain_ir = plan.explain();
+        let mut graph = Graph::<String, String>::new();
+        let mut nodes = HashMap::new();
+        build_graph_from_pretty(&explain_ir, &mut graph, &mut nodes, None);
+        let dot = Dot::with_config(&graph, &[Config::EdgeNoLabel]);
+        dot.to_string()
+    }
+}
+
+fn build_graph_from_pretty(
+    pretty: &Pretty<'_>,
+    graph: &mut Graph<String, String>,
+    nodes: &mut HashMap<String, NodeIndex>,
+    parent_label: Option<&str>,
+) {
+    if let Pretty::Record(r) = pretty {
+        let mut label = String::new();
+        label.push_str(&r.name);
+        if !r.fields.is_empty() {
+            label.push_str(" fields: ");
```

Comment:
The alternative is to add each field to a new line, this is much closer to the desired pictures provided
```
        for (k, v) in r.fields.iter() {
            label.push_str("\n");
            label.push_str(&k);
            label.push_str(": ");
            label.push_str(&serde_json::to_string(&PrettySerde(v.clone())).expect("failed to serialize plan to dot"));
        }
```
However, the visualized output does not look great in terms of alignment
![image](https://github.com/user-attachments/assets/73c0cce3-fc96-4cdf-9363-6b480cc94cd2)

Reply:
see: https://github.com/risingwavelabs/risingwave/pull/19446/files#r1849893022